staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -394,7 +574,7 @@ func diffListsOfScalars(original, modified []interface{}, diffOptions DiffOption
 		}
 	}
 
-	return addList, deletionList, nil
+	return addList, uniqifyScalars(deletionList), nil
```

Comment:
I didn't find that surprising (neither the spelling nor the name). It comes pretty naturally as an extension of [uniq](http://pubs.opengroup.org/onlinepubs/9699919799/utilities/uniq.html).